src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -249,12 +260,25 @@ case class AppDefinition(
       else
         UnreachableStrategy.default(isResident)
 
+    val executorRes: Option[Resources] = if (this.executorResources.isEmpty && executorResourcesMap.isEmpty)
+      None
+    else
+      Some(Resources(
+        cpus = executorResourcesMap.getOrElse(
+          Resource.CPUS, this.executorResources.getOrElse(AppDefinition.MinimalExecutorResources).cpus),
+        mem = executorResourcesMap.getOrElse(
+          Resource.MEM, this.executorResources.getOrElse(AppDefinition.MinimalExecutorResources).mem),
+        disk = executorResourcesMap.getOrElse(
+          Resource.DISK, this.executorResources.getOrElse(AppDefinition.MinimalExecutorResources).disk)
+      ))
+
```

Comment:
this whole `executorResourcesMap` thing...  lines 228-232 and 263-274...   There is no external reference to it that I see...  As this is all internal... this seems like a complicated way of managing these details.   The nested Somes with defaults ... 

why don't we just default the map with defaults at creation, then add what is provided by the protob?
it would simplify this to just getting the values.

Reply:
I also don't care for the shorten name `executorRes`

it is an easy replacement for `executorResourcesMap.isEmpty` to `proto.getExecutorResourcesList.isEmpty`

I don't see the value of `this.executorResources.isEmpty` as this is in `mergeFromProto` and there is no way `executorResources` could have been set... which makes it 1 logical jump away from `DefaultExecutorResources` which is `None`

Reply:
looking closer it looks like you are trying to maintain the same code as `resourcesMap`... which totally makes sense... however It seems more challenging from a readability standpoint to read this:
```
   cpus = executorResourcesMap.getOrElse(
          Resource.CPUS, executorResources.getOrElse(AppDefinition.MinimalExecutorResources).cpus),
```
and this:
```
        cpus = resourcesMap.getOrElse(Resource.CPUS, this.resources.cpus),
```

if we can have the "Else" as the default vs another "Else"... it would really improve readability IMO.